src/DsaBottomUp.cc

```diff
@@ -290,56 +369,126 @@ bool BottomUpAnalysis::runOnModule(Module &M, GraphMap &graphs) {
-        LOG("dsa-bu", llvm::errs() << "\tCallee collapsed: " << calleeG.numCollapsed() << ", caller collapsed:\t" << callerG.numCollapsed() << "\n");
-
-        cloneAndResolveArguments(dsaCS, calleeG, callerG, m_noescape);
-        LOG("dsa-bu", llvm::errs() << "\tCaller size after clone: " << callerG.numNodes() << ", collapsed: " << callerG.numCollapsed() << "\n");
+        LOG("dsa-bu", llvm::errs() << "BU #" << cnt << ": "
+                                   << dsaCS.getCaller()->getName() << " <- "
+                                   << dsaCS.getCallee()->getName() << "\n");
+        LOG("dsa-bu", llvm::errs()
+                          << "\tCallee size: " << calleeG.numNodes()
+                          << ", caller size:\t" << callerG.numNodes() << "\n");
+        LOG("dsa-bu", llvm::errs()
+                          << "\tCallee collapsed: " << calleeG.numCollapsed()
+                          << ", caller collapsed:\t" << callerG.numCollapsed()
+                          << "\n");
+
+        cloneAndResolveArguments(dsaCS, calleeG, callerG, m_noescape,
+                                 copiedCallSites);
+        LOG("dsa-bu", llvm::errs()
+                          << "\tCaller size after clone: " << callerG.numNodes()
+                          << ", collapsed: " << callerG.numCollapsed() << "\n");
       }
 
-      if (m_computeSimMap) {
-        // -- store the simulation maps from the SCC
-        for (auto &callRecord : *cgn) {
-          ImmutableCallSite CS(callRecord.first);
-          DsaCallSite dsaCS(CS);
-          const Function *callee = dsaCS.getCallee();
-          if (!callee || callee->isDeclaration() || callee->empty())
-            continue;
-
-          assert(graphs.count(dsaCS.getCaller()) > 0);
-          assert(graphs.count(dsaCS.getCallee()) > 0);
-
-          Graph &callerG = *(graphs.find(dsaCS.getCaller())->second);
-          Graph &calleeG = *(graphs.find(dsaCS.getCallee())->second);
-
-          SimulationMapperRef sm(new SimulationMapper());
-          bool res = Graph::computeCalleeCallerMapping(dsaCS, calleeG, callerG,
-                                                       *sm, do_sanity_checks);
-          if (!res) {
-            // continue;
-            llvm_unreachable("Simulation mapping check failed");
-          }
-          m_callee_caller_map.insert(
-              std::make_pair(dsaCS.getInstruction(), sm));
-
-          if (do_sanity_checks) {
-            // Check the simulation map is a function
-            if (!sm->isFunction())
-              errs() << "ERROR: simulation map for " << *dsaCS.getInstruction()
-                     << " is not a function!\n";
-            // Check that all nodes in the callee are mapped to one
-            // node in the caller graph
-            checkAllNodesAreMapped(*callee, calleeG, *sm);
-          }
-        }
-      }
+      /////
+      /// FIXME: Commenting this code breaks the SAS'17 code.
```

Comment:
`#def` or some option would be better than commenting the code like that.
Would be good that the old functionality is preserved, even if it only works when there are no indirect call sites in the input file